src/AzureIoTHub.Portal.Infrastructure/Migrations/20221028152842_Store edge device connection state.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) CGI France. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable disable
+
+namespace AzureIoTHub.Portal.Infrastructure.Migrations
+{
+    using Microsoft.EntityFrameworkCore.Migrations;
+
+    public partial class Storeedgedeviceconnectionstate : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            _ = migrationBuilder.AddColumn<string>(
+                name: "ConnectionState",
+                table: "EdgeDevices",
+                type: "text",
+                nullable: true,
+                defaultValue: "");
+
+            _ = migrationBuilder.Sql("UPDATE \"EdgeDevices\" SET \"ConnectionState\" = ''");
+
+            _ = migrationBuilder.AlterColumn<string>(
+                name: "ConnectionState",
+                table: "EdgeDevices",
+                nullable: false);
```

Comment:
@Sben65 , @audserraCGI and @ChristopheRib63 

Please see here how we can add a new column that should be not nullable: 

1. Add the column as nullable
2. Create a script to migrate existing data (set the default value to each rows)
3. Alter the column to set it to not nullable...

I suspect that this was the issue you had before in #1464 ...